src/ctia/bundle/core.clj

```diff
@@ -75,30 +79,34 @@
       (seq filtered-ext-ids) (assoc :external_ids filtered-ext-ids))))
 
 (s/defn all-pages
-  "Retrieves all external ids using pagination."
+  "Retrieves all entities by external ids using pagination."
   [entity-type
    external-ids
-   auth-identity
+   auth-identity :- auth/AuthIdentity
    get-store :- GetStoreFn]
   (loop [ext-ids external-ids
          entities []]
-    (let [query {:all-of {:external_ids ext-ids}}
-          paging {:limit find-by-external-ids-limit}
-          {results :data
-           {next-page :next} :paging} (-> (get-store entity-type)
-                                          (store/list-records
-                                           query
-                                           (auth/ident->map auth-identity)
-                                           paging))
-          acc-entities (into entities results)
-          matched-ext-ids (into #{} (mapcat :external_ids results))
-          remaining-ext-ids (remove matched-ext-ids ext-ids)]
-      (if next-page
-        (recur remaining-ext-ids acc-entities)
-        acc-entities))))
+    (if (empty? ext-ids)
+      entities
```

Comment:
Something about `all-pages` seems wrong based on my reading.

It checks `next-page` but doesn't use pagination and then recurs to the first page of a different query.

Say an external id is mapped to 2 entities with a page limit of 1. First, we miss the 2nd entity altogether since we recur with an empty `ext-ids`. And second, since `ext-ids` is empty, we make an empty query.

Reply:
Good to know, reverted.